src/lib.rs

```diff
@@ -222,10 +222,13 @@ impl TwoByteWM {
     }
 
     /// Returns an iterator over non-overlapping matches.
-    pub fn find<'a, 'b>(&'a self, haystack: &'b str) -> Matches<'a, 'b> {
+    pub fn find<'a, 'b, P>(&'a self, haystack: &'b P) -> Matches<'a, 'b>
```

Comment:
You should take ownership of `P` here, i.e. remove the `&`.

The deref traits will make this work with anything relevant, because `as_ref` has a reference receiver. For example:

```rust
fn take_it<B: AsRef<[u8]>>(b: B) -> usize {
    b.as_ref().len()
}

fn main() {
    // &str
    take_it("foo");

    // Vec<u8>
    let vec: Vec<u8> = vec![];
    take_it(vec.clone());

    // &Vec<u8>
    take_it(&vec);

    // &[u8]
    take_it(vec.as_slice());

    // String
    take_it("foo".to_string());
}
```

You then don't need to change the API and all the test code should keep working as before.

Reply:
I had to propagate this a bit further into `Matches`, but in the end it did work out. PTAL.